board/gpio.md

```diff
@@ -3,70 +3,97 @@ GPIO API
 
 The GPIO (General Purpose Input & Output) API supports digital pins.
 
-The API object
---------------
-GPIO pin functionality is exposed by the [`GPIO`](#gpio) object that can be obtained by using the [gpio() method of the `Board` API](./README.md/#gpio). See also the [Web IDL](./webidl.md).
-
 On certain boards GPIO pins may be grouped into ports (e.g. 8, 16 or 32 pins), read and written as registers by the controller.
 
 On certain boards, analog pins can also be used as GPIO.
 
-<a name="gpio"></a>
-### The `GPIO` interface
-Represents the properties and methods that expose GPIO functionality. The `GPIO` object implements the [`EventEmitter`](../README.md/#events) interface, and extends the [`Pin`](./README.md/#pin) object.
+<a name="apiobject"></a>
+### The GPIO API object
+When requiring `"gpio"`, the following steps are run:
+- If there is no permission for using the functionality, throw `SecurityError`.
+- If the AIO functionality is not supported on the board, throw `"NotSupportedError"`.
+- Return an object that implements the following methods.
+
+| Method              | Description      |
+| ---                 | ---              |
+| [`open()`](#open)   | open GPIO pin    |
+| [`port()`](#port)   | open GPIO port   |
+
+See also the [Web IDL](./webidl.md) definition.
+
+The following dictionary is used for initializing GPIO pins and ports.
+<a name="gpiooptions"></a>
 
 | Property   | Type   | Optional | Default value | Represents |
 | ---        | ---    | ---      | ---           | ---        |
-| `pin`      | String or Number | no | `undefined`   | board name for the pin |
+| `pin `     | String or Number | no | `undefined`   | pin name |
+| `mapping`  | String | no | `"system"`   | pin mapping |
```

Comment:
This would be too Zephyr-specific. In general, I wanted to record the developer choice (or whether to have a preference at all) whether they want to look at Zephyr docs or the board labels.

As for implementation docs, that's better avoided: then those should be spec'd and included in board-specific APIs in iot-js-api (and Zephyr.js also) -- which we also wanted to avoid.

Otherwise, in general, implementations could do the mapping themselves, so in Zephyr case instead of saying `GPIO_9.15` they could just say 15 as well - but then that would require an iot-js-api specific doc, which we wanted to avoid.